tests/mock_network/node_ageing.rs

```diff
@@ -334,36 +291,77 @@ fn node_relocated(
     source_prefix: &Prefix<XorName>,
     target_prefix: &Prefix<XorName>,
 ) -> bool {
-    let node_name = nodes[node_index].name();
-    for node in nodes {
-        let prefixes = node.inner.prefixes();
-
-        let in_source = prefixes
-            .iter()
-            .filter(|prefix| prefix.is_compatible(source_prefix))
-            .any(|prefix| {
-                // TODO: check all members, not just elders.
-                node.inner.section_elders(prefix).contains(&node_name)
-            });
-        if in_source {
+    let relocated_id = nodes[node_index].id();
+
+    for node in nodes_with_prefix(nodes, source_prefix) {
+        if !node.inner.is_elder() {
+            continue;
+        }
+
+        if node.inner.is_peer_our_member(&relocated_id) {
+            trace!(
+                "Node {} is member of the source section {:?} according to {}",
+                relocated_id.name(),
+                source_prefix,
+                node.name()
+            );
             return false;
         }
+    }
 
-        let in_target = prefixes
-            .iter()
-            .filter(|prefix| prefix.is_compatible(target_prefix))
-            .any(|prefix| {
-                // TODO: check all members, not just elders.
-                node.inner.section_elders(prefix).contains(&node_name)
-            });
-        if !in_target {
+    for node in nodes_with_prefix(nodes, target_prefix) {
+        if !node.inner.is_elder() {
+            continue;
+        }
+
+        let node_prefix = node.inner.our_prefix().unwrap();
+        if node_prefix.is_extension_of(target_prefix) && !node_prefix.matches(relocated_id.name()) {
+            // Target section has split and the relocated node is in the other sub-section than this node.
+            continue;
+        }
+
+        if !node.inner.is_peer_our_member(&relocated_id) {
+            trace!(
+                "Node {} is not member of the target section {:?} according to {}",
+                relocated_id.name(),
+                target_prefix,
+                node.name()
+            );
             return false;
         }
     }
 
     true
 }
 
+fn verify_node_relocated(
+    nodes: &[TestNode],
+    node_index: usize,
+    source_prefix: &Prefix<XorName>,
+    target_prefix: &Prefix<XorName>,
+) {
+    assert!(
+        node_relocated(nodes, node_index, source_prefix, target_prefix),
+        "Node {} did not get relocated from {:?} to {:?}",
+        nodes[node_index].name(),
+        source_prefix,
+        target_prefix
+    );
+}
+
+// Returns the age counter of the node with the given name.
+fn node_age_counter(nodes: &[TestNode], name: &XorName) -> u32 {
+    if let Some(counter) = nodes
+        .iter()
+        .filter_map(|node| node.inner.member_age_counter(name))
```

Comment:
I thought a node's age counter shall be from its own perspective.
However this code block seems scanning through all members' perspective?

Reply:
No, it's from the section perspective, because only elders store the age counters.